pyproject.toml

```diff
@@ -28,41 +28,153 @@ classifiers = [
     "Natural Language :: English",
     "Programming Language :: C++",
     "Programming Language :: Python",
-    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
     "Topic :: Scientific/Engineering",
     "Topic :: Scientific/Engineering :: Mathematics",
     "Topic :: Scientific/Engineering :: Physics",
 ]
 dependencies = [
     "importlib-resources; python_version<'3.9'",
     "numpy",
-    "platformdirs>=2.2.0",
+    "platformdirs>=2.2",
     "pytools>=2024.1.5",
 ]
 
 [project.optional-dependencies]
 oclgrind = [
-    "oclgrind_binary_distribution>=18.3",
+    "oclgrind-binary-distribution>=18.3",
 ]
 pocl = [
-    "pocl_binary_distribution>=1.2",
+    "pocl-binary-distribution>=1.2",
 ]
 test = [
-    "Mako",
-    "pytest>=7.0.0",
+    "flake8",
+    "flake8-bugbear",
+    "flake8-comprehensions",
+    "flake8-isort",
+    "flake8-quotes",
+    "isort",
+    "mako",
+    "mypy",
+    "pep8-naming",
+    "pylint",
+    "pytest>=7",
 ]
 
 [project.urls]
-Homepage = "https://mathema.tician.de/software/pyopencl"
 Documentation = "https://documen.tician.de/pyopencl"
+Homepage = "https://mathema.tician.de/software/pyopencl"
 Repository = "https://github.com/inducer/pyopencl"
 
+[tool.scikit-build]
+sdist.exclude = [
+    ".mypy_cache",
+    ".ci",
+    ".github",
+    ".conda-ci-build-configure.sh",
+    "doc/upload-docs.sh",
+    ".editorconfig",
+    "TODOs",
+    "run-*.sh",
+    "setup.cfg",
+]
+
+[tool.aksetup]
+disable-editable-pip-install = true
+enable-flake8-isort = true
+enable-flake8-bugbear = true
+
+[tool.flake8]
+max-line-length = 85
+inline-quotes = "double"
+docstring-quotes = "double"
+multiline-quotes = "double"
+select = [
+    "B",    # flake8-bugbear
+    "C",    # flake8-comprehensions
+    "D",    # pydocstyle
+    "E",    # pycodestyle
+    "F",    # pyflakes
+    "I",    # flake8-isort
+    "N",    # pep8-naming
+    "Q",    # flake8-quotes
+    "W",    # pycodestyle
+]
```

Comment:
I added this too, but I'm not very convinced, more for a small discussion.

I thought about it mainly for two reasons:
* removing the `enable-isort` (and others) grepping in `ci-support`: with this it can just install all the plugins it wants and the project will select the ones it wants.
* if we move to `ruff`, all these are bundled anyway and we'll need to select the ones we want.
* *it's unlikely at this point, but this would also guard against new added error codes.*

Reply:
I think I prefer this, even. Getting ci-support to understand this won't be too much of a lift. I like it mostly because it allows us to avoid making up our own idiotic config bits.

Reply:
> I like it mostly because it allows us to avoid making up our own idiotic config bits.

Yeah, that too! We can leave it in then, and slowly teach `ci-support` and everybody else to slowly migrate.

Reply:
Why do we even need this? If we have the tools listed in `project.optional-dependencies.test`, we could just use that.

Reply:
Hmm. I guess the "ruff" argument stands. :/ How close is ruff a this point? Should we just switch to it?

Reply:
Hm, yeah, that's a good point. Sounds easier to standardize on some optional dependencies group name and just work with that if it exists!

I kind of like that better.

Reply:
> Hmm. I guess the "ruff" argument stands. :/ How close is ruff a this point? Should we just switch to it?

It doesn't have a lot of the fancier pylint checks that actually execute the code (or whatever they do), but everything else is there and seems to be working very nicely!

I would say it's pretty solid at this point (since they got bored and re-implemented `black` as well :))

Reply:
Here's a list of the pylint checks that are implemented
https://github.com/astral-sh/ruff/issues/970

Reply:
Alright, let's say ruff is the future then, replacing flake8+plugins. Pylint can live for the moment.